src/aggregation/metric/extended_stats.rs

```diff
@@ -0,0 +1,1177 @@
+    sigma: f64,
+}
+
+impl Default for IntermediateExtendedStats {
+    fn default() -> Self {
+        Self {
+            intermediate_stats: IntermediateStats::default(),
+            sum_of_squares: 0.0,
+            sum_of_squares_elastic: 0.0,
+            delta_sum_for_squares_elastic: 0.0,
+            mean: 0.0,
+            sigma: 2.0,
+        }
+    }
+}
+
+impl IntermediateExtendedStats {
+    /// Creates a new IntermediateExtendedStats using an option
+    /// containing the sigma to be used for calculating bound values.
+    pub fn with_sigma(sigma: Option<f64>) -> Self {
+        Self {
+            intermediate_stats: IntermediateStats::default(),
+            sum_of_squares: 0.0,
+            sum_of_squares_elastic: 0.0,
+            delta_sum_for_squares_elastic: 0.0,
+            mean: 0.0,
+            sigma: sigma.unwrap_or(2.0),
+        }
+    }
+    /// Merges the other stats intermediate result into self.
+    pub fn merge_fruits(&mut self, other: IntermediateExtendedStats) {
+        if other.intermediate_stats.count == 0 {
+            return;
+        }
+        if self.intermediate_stats.count == 0 {
+            let _ = mem::replace(self, other);
+            return;
+        }
+        let new_count = self.intermediate_stats.count + other.intermediate_stats.count;
+        let delta = other.mean - self.mean;
+        self.sum_of_squares += other.sum_of_squares
+            + delta
+                * delta
+                * self.intermediate_stats.count as f64
+                * other.intermediate_stats.count as f64
+                / new_count as f64;
+        self.mean = (self.intermediate_stats.sum as f64 + other.intermediate_stats.sum as f64)
+            / new_count as f64;
+        self.sum_of_squares_elastic += other.sum_of_squares_elastic;
+        self.delta_sum_for_squares_elastic += other.delta_sum_for_squares_elastic;
+        self.intermediate_stats
+            .merge_fruits(other.intermediate_stats);
+    }
+
+    /// Computes the final stats value.
+    pub fn finalize(&self) -> Box<ExtendedStats> {
+        let (min, max, avg, sum_of_squares) = if self.intermediate_stats.count == 0 {
+            (None, None, None, None)
+        } else {
+            (
+                Some(self.intermediate_stats.min),
+                Some(self.intermediate_stats.max),
+                Some(self.mean),
+                Some(self.sum_of_squares_elastic),
+            )
+        };
+        let (variance, variance_sampling) = if self.intermediate_stats.count <= 1 {
+            (None, None)
+        } else {
+            (
+                Some(self.sum_of_squares / self.intermediate_stats.count as f64),
+                Some(self.sum_of_squares / (self.intermediate_stats.count - 1) as f64),
+            )
+        };
+        let std_deviation = variance.map(|v| v.sqrt());
+        let std_deviation_sampling = variance_sampling.map(|v| v.sqrt());
+        let std_deviation_bounds = if std_deviation.is_none() {
```

Comment:
can you use `if let Some(std_deviation, std_deviation_sampling) = ` instead of `is_some` + unwrap?